src/test/java/com/commercetools/sync/producttypes/utils/ProductTypesUpdateActionUtilsTest.java

```diff
@@ -0,0 +1,129 @@
+        when(old.getDescription()).thenReturn("description 1");
+        when(old.getAttributes()).thenReturn(oldAttributeDefinitions);
+
+        newSame = ProductTypeDraft.ofAttributeDefinitionDrafts(
+                "key1",
+                "name1",
+                "description 1",
+                sameAttributeDefinitionDrafts
+        );
+
+        newDifferent = ProductTypeDraft.ofAttributeDefinitionDrafts(
+                "key2",
+                "name2",
+                "description 2",
+                sameAttributeDefinitionDrafts
+        );
+
+        newWithNullValues = ProductTypeDraft.ofAttributeDefinitionDrafts(
+                null,
+                "name3",
+                "description 3",
+                sameAttributeDefinitionDrafts
+        );
+    }
+
+    @Test
+    public void buildSetKeyAction_WithDifferentValues_ShouldReturnAction() {
+        final Optional<UpdateAction<ProductType>> result = buildSetKeyAction(old, newDifferent);
+        assertThat(result).isNotNull();
+        assertThat(result.isPresent()).isTrue();
+        assertThat(result.get()).isExactlyInstanceOf(SetKey.class);
+        assertThat(((SetKey) result.get()).getKey()).isEqualTo(newDifferent.getKey());
+    }
+
+    @Test
+    public void buildSetKeyAction_WithSameValues_ShouldReturnEmptyOptional() {
+        assertNoUpdatesForSameValues(ProductTypesUpdateActionUtils::buildSetKeyAction);
+    }
+
+    @Test
+    public void buildChangeNameAction_WithDifferentValues_ShouldReturnAction() {
+        final Optional<UpdateAction<ProductType>> result = buildChangeNameAction(old, newDifferent);
+        assertThat(result).isNotNull();
+        assertThat(result.isPresent()).isTrue();
+        assertThat(result.get()).isExactlyInstanceOf(ChangeName.class);
+        assertThat(((ChangeName) result.get()).getName()).isEqualTo(newDifferent.getName());
+    }
+
+    @Test
+    public void buildChangeNameAction_WithSameValues_ShouldReturnEmptyOptional() {
+        assertNoUpdatesForSameValues(ProductTypesUpdateActionUtils::buildChangeNameAction);
+    }
+
+    @Test
+    public void buildChangeDescriptionAction_WithDifferentValues_ShouldReturnAction() {
+        final Optional<UpdateAction<ProductType>> result = buildChangeDescriptionAction(old, newDifferent);
+        assertThat(result).isNotNull();
+        assertThat(result.isPresent()).isTrue();
+        assertThat(result.get()).isExactlyInstanceOf(ChangeDescription.class);
+        assertThat(((ChangeDescription) result.get()).getDescription()).isEqualTo(newDifferent.getDescription());
+    }
+
+    @Test
+    public void buildChangeDescriptionAction_WithSameValues_ShouldReturnEmptyOptional() {
+        assertNoUpdatesForSameValues(ProductTypesUpdateActionUtils::buildChangeDescriptionAction);
+    }
+
+    private void assertNoUpdatesForSameValues(final BiFunction<ProductType, ProductTypeDraft,
+            Optional<UpdateAction<ProductType>>> buildFunction) {
```

Comment:
Nice that you created a util and reused it but actually here you can also use the `AbstractOptionalAssert`

````java
assertThat(result).isEmpty();
````

in which case it would be a one liner that you can directly use without maybe needing the `assertNoUpdatesForSameValues` :see_no_evil:

Reply:
https://github.com/commercetools/commercetools-sync-java/pull/287/commits/fe004ec241788231f035ae6c439db07f7f322642